test/notification/channel.go

```diff
@@ -0,0 +1,168 @@
+package notification
```

Comment:
Any particular reason why these files aren't in `_test.go` files.

More importantly: why aren't they next to the code they are testing?

Reply:
To clarify: if (for some reason unknown to me) it does make sense to have a separate test package, I'd still prefer having them closer to the original notification code.

I definitely don't like having a high level test package (not a huge fan of e2e either, but there is a good reason for that)

Reply:
> Any particular reason why these files aren't in _test.go files.

There is a single test file called `notification_test.go` which includes all the tests covered by this package.
There are multiple test "suites" in dedicated files like `channel.go` which cannot be directly tested as they require inicialization. Initialization is done in `notification_test.go` and the test "suites" are run from there.

Not particualrly keen on this structure, but my focus was on having test coverage and so much how the tests are structured. But I personally prefer to have a discussion around how we write/run integration/e2e tests just to have some guidelines.

> More importantly: why aren't they next to the code they are testing?

Again, this is something we need to chat about. I perfer unit tests stored next to the code, but having separate integration and e2e test packages. But I ok with whatever structure the team happy with.

Reply:
I prefer to keep it as is until there is a decision how we want to sturucture our tests. Afterwards I'll make this test package aligned with the guidlines.